ft/src/lib.rs

```diff
@@ -74,12 +80,35 @@ impl Contract {
         let mut this = Self {
             token: FungibleToken::new(b"a".to_vec()),
             metadata: LazyOption::new(b"m".to_vec(), Some(&metadata)),
+            burn_token_account
         };
         this.token.internal_register_account(owner_id.as_ref());
         this.token.internal_deposit(owner_id.as_ref(), total_supply.into());
         this
     }
 
+    pub fn burn_token( 
+        &mut self,
+        account_id: ValidAccountId,
```

Comment:
We don't want to burn tokens in _whatever_ account. We only burn token in the account `self.burn_token_account`.

Reply:
Okay, let me update

Reply:
LGTM